app/server.py

```diff
@@ -63,6 +63,8 @@ def search(search_engine):
         if db['queries'].find({query: query}).limit(1) is False:
             db['queries'].insert(
                 {"query": query, "engine": engine, "qformat": qformat})
```

Comment:
And also, we need to check for tiimestamp as @cclauss said.
We also need to remove/update the outdated results.

Regarding the format of data, I think we can just store data in json in mongodb, and convert it ondemand, as requested in the request.